### PR TITLE
Update RenderControl.cpp

### PR DESCRIPTION
**Pull request by novhy**
this will make some apps opened correctly,
like google chrome

### CODE REVIEW

File: src/anbox/graphics/emugl/RenderControl.cpp [hunk 1]
```diff
@@ -120,6 +120,12 @@ static EGLint rcGetGLString(EGLenum name, void* buffer, EGLint bufferSize) {
   // which aren't met.
   if (name == GL_VERSION)
     result = "OpenGL ES 2.0";
+  if (name == GL_VENDOR)
+    result = "Google ("+result+")";
```
Comment (morphis) [inline src/anbox/graphics/emugl/RenderControl.cpp:124]: Can we add spaces in here to honor the overall coding style?

File: src/anbox/graphics/emugl/RenderControl.cpp [hunk 2]
```diff
@@ -120,6 +120,12 @@ static EGLint rcGetGLString(EGLenum name, void* buffer, EGLint bufferSize) {
...
+  if (name == GL_RENDERER)
+    result = "Android Emulator OpenGL ES ("+result+")";
```
Comment (morphis) [inline src/anbox/graphics/emugl/RenderControl.cpp:126]: See above

**Review by morphis — CHANGES_REQUESTED** [review]
Do you have a list of apps which this helps to make working?

**Comment by Niek** [comment]
I was hoping this would fix the issues in https://github.com/anbox/anbox/issues/1637 since you mentioned Chrome, but with this PR applied I still get these errors:

```bash
12-02 09:41:48.547  1012  1033 E EGL_emulation: tid 1033: eglCreateContext(919): error 0x3005 (EGL_BAD_CONFIG)
12-02 09:41:48.547  1012  1033 E chromium: [ERROR:gl_context_egl.cc(248)] eglCreateContext failed with error EGL_BAD_CONFIG
12-02 09:41:48.547  1012  1033 E chromium: [ERROR:gpu_channel_manager.cc(749)] ContextResult::kFatalFailure: Failed to create shared context for virtualization.
12-02 09:41:48.547  1012  1033 E chromium: [ERROR:shared_image_stub.cc(470)] SharedImageStub: unable to create context
12-02 09:41:48.547  1012  1033 E chromium: [ERROR:gpu_channel.cc(449)] GpuChannel: Failed to create SharedImageStub
```

**Comment by novhy** [comment]
im build anbox version from
https://github.com/anbox/anbox/commit/6386cf57bc3a17ff9b89c22f486e459d4c141a6b


patched for ubuntu touch
and android img from

http://anbox.postmarketos.org/

arm64 version

chrome works, gapps works

but after editing /system/etc/permissions

added standard android features

all app in playstore showup

but, chrome,firefox becoming useless again


i think chrome,firefox blocking some opengles extention or some what on device that detected as emulator

just try with standard android img for anbox and that patch

**Comment by Niek** [comment]
> im build anbox version from
> [6386cf5](https://github.com/anbox/anbox/commit/6386cf57bc3a17ff9b89c22f486e459d4c141a6b)
> 
> patched for ubuntu touch
> and android img from
> 
> http://anbox.postmarketos.org/
> 
> arm64 version
> 
> chrome works, gapps works
> 
> but after editing /system/etc/permissions
> 
> added standard android features
> 
> all app in playstore showup
> 
> but, chrome,firefox becoming useless again
> 
> i think chrome,firefox blocking some opengles extention or some what on device that detected as emulator
> 
> just try with standard android img for anbox and that patch

Which "ubuntu touch patch" are you referring to? I tested anbox (latest master) + this PR + copied android/media/* to /system/etc + http://anbox.postmarketos.org/ image on x86 and I can open Chrome, but cannot load any page.

**Comment by novhy** [comment]
ubuntu touch patch from anbox ubport github repo,

hmm, chrome open and  adb logs is about segmentation fault in glgetshaderprecisionformat?

**Comment by Niek** [comment]
@novhy do you have a repo with the source you're building? I have issues merging the ubports repo changes to upstream cleanly. I still haven't been able to run a recent Chrome correctly. It starts up fine, but no page renders (EGL_BAD_CONFIG errors).

**Comment by kokhui** [comment]
> im build anbox version from
> [6386cf5](https://github.com/anbox/anbox/commit/6386cf57bc3a17ff9b89c22f486e459d4c141a6b)
> 
> patched for ubuntu touch
> and android img from
> 
> http://anbox.postmarketos.org/
> 
> arm64 version
> 
> chrome works, gapps works
> 
> but after editing /system/etc/permissions
> 
> added standard android features
> 
> all app in playstore showup
> 
> but, chrome,firefox becoming useless again
> 
> i think chrome,firefox blocking some opengles extention or some what on device that detected as emulator
> 
> just try with standard android img for anbox and that patch

Hi, can I check does your gapps works in --single window mode? It keep popping up prompt that google services is stopping when I run in that mode.

Using arm64 ver from anbox.postmarketos.org and gapps arm from opengapps.org/. (Running on AWS ec2 arm 64bit instance)

**Comment by Shatur95** [comment]
@novhy, what kind of Ubports patches have you applied?

**Comment by novhy** [comment]
screenshot20210314_003652703

sorry everyone, that patch is missing something,
i will upload my source later, or finding what cause, and pull request to this repo

**Comment by Shatur95** [comment]
@novhy, it would be awesome!

**Comment by Shatur95** [comment]
@novhy, any news about it? Sorry for bothering.

**Pull request closed without merging**